### PR TITLE
Added terminalio font support and scaling

### LINKED ISSUE #72: Default param for text_font
Currently text_font on PyPortal class is set to None. Is there a chance we could utilize termalio font for a default font?

### PR DESCRIPTION
**Pull request by makermelissa**
Fixes #72.

### CODE REVIEW

**Comment by makermelissa** [comment]
The reason I added scaling into this PR (backported from MatrixPortal) is because the terminalio font at default size is very small.

**Comment by tannewt** [comment]
Could we share the graphics bits of the two portal libraries instead of backporting between the two?

**Comment by makermelissa** [comment]
Maybe, but the graphics are dealt with quite differently due to the different display sizes. For instance, I have a scrolling type for the Matrix that isn't needed with the PyPortal.

File: adafruit_pyportal.py [hunk 1]
```diff
@@ -399,6 +407,9 @@ def __init__(
                 self._text_wrap[i] = text_wrap[i]
                 self._text_maxlen[i] = text_maxlen[i]
                 self._text_transform[i] = text_transform[i]
+                if not isinstance(text_scale[i], (int, float)) or text_scale[i] < 1:
```
Comment (FoamyGuy) [inline adafruit_pyportal.py:410]: I tested these changes and I am getting and I'm getting an exception from this line:
```
Traceback (most recent call last):
  File "code.py", line 80, in <module>
  File "/lib/adafruit_pyportal.py", line 410, in __init__
TypeError: 'int' object is not subscriptable
```
I'm using the code from the quotes example here: https://learn.adafruit.com/adafruit-pyportal/parsing-json. I tried with the code as-is, and with the new `text_scale` parameter added, I'm getting same results both ways.

Comment (FoamyGuy) [reply adafruit_pyportal.py:410]: I was able to get past this error by adding a check for `text_scale` type check inside the body of the if statement at line 368 of the modified code. Now mine looks like:
```python
if isinstance(text_position[0], (list, tuple)):
    num = len(text_position)
    if not text_wrap:
        text_wrap = [0] * num
    if not text_maxlen:
        text_maxlen = [0] * num
    if not text_transform:
        text_transform = [None] * num
    if isinstance(text_scale, int):  # new check for text scale
        text_scale = [text_scale] * num
```

**Comment by tannewt** [comment]
> Maybe, but the graphics are dealt with quite differently due to the different display sizes. For instance, I have a scrolling type for the Matrix that isn't needed with the PyPortal.

kk, maybe it's the network stuff we should be sharing. It makes sense the graphics would be different.

**Comment by makermelissa** [comment]
Thanks @FoamyGuy, I updated the code.

**Review by FoamyGuy — APPROVED** [review]
Tested with the newest commits and it's working well now.

Looks good to me. Thanks for this new feature @makermelissa

**Pull request merged**